### ISSUE TITLE
messageId doc examples throw during testing.

### ISSUE THREAD

**Issue by betaorbust** [eslint/eslint#9889]
**Tell us about your environment**

* **ESLint Version:** 4.15+
* **Node Version:** 8.9.3
* **npm Version:** 5.5.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
N/A

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**
Ran the code provided in the [messageId](https://eslint.org/docs/developer-guide/working-with-rules#messageids) section of the Working with Rules page.
Recreated below: 

lib/rules/no-foo.js
```js
'use strict';
module.exports = {
    meta: {
        messages: {
            avoidName: "Avoid using variables named '{{ name }}'"
        }
    },
    create(context) {
        return {
            Identifier(node) {
                if (node.name === "foo") {
                    context.report({
                        node,
                        messageId: "avoidName",
                        data: {
                            name: "foo",
                        }
                    });
                }
            }
        };
    }
};
```

tests/lib/rules/no-foo.js
*Updated require path which was `var rule = require('../../../lib/rules/no-insecure-random');` which seems unrelated to this example.*
```js
'use strict';
// In your tests:
var rule = require('../../../lib/rules/no-foo');
var RuleTester = require('eslint').RuleTester;

var ruleTester = new RuleTester();
ruleTester.run('my-rule', rule, {
  valid: ['bar', 'baz'],

  invalid: [
    {
      code: 'foo',
      errors: [
        {
          messageId: 'foo',
        }
      ]
    }
  ]
});
```

**What did you expect to happen?**
It should work out of the box more or less.

**What actually happened? Please include the actual, raw output from ESLint.**
```bash
 1 failing

  1) my-rule invalid foo:
     AssertionError [ERR_ASSERTION]: Invalid messageId 'foo'. Expected one of ['avoidName'].
      at testInvalidTemplate (node_modules\eslint\lib\testers\rule-tester.js:508:40)
      at Context.RuleTester.it (node_modules\eslint\lib\testers\rule-tester.js:581:25)
```

**Comment by platinumazure** [comment]
Hi @betaorbust, thanks for the issue! I agree 100%, those docs need tweaking. Would you like to submit a pull request?

**Comment by betaorbust** [comment]
Unfortunately, while putting together my PR for this, I ran into a much bigger problem in how the `messageId` feature currently works.

See https://github.com/eslint/eslint/issues/9890 for more details.
